include/aws/io/socket.h

```diff
@@ -172,10 +185,15 @@ AWS_IO_API void aws_socket_clean_up(struct aws_socket *socket);
  * In TCP, LOCAL and VSOCK this function will not block. If the return value is successful, then you must wait on the
  * `on_connection_result()` callback to be invoked before using the socket.
  *
+ * The function will failed with error if the endpoint is invalid, except for Apple Network Framework. In Apple network
+ * framework, as connect is an async api, we would not know if the local endpoint is valid until we have the connection
+ * state returned in callback. The error will returned in `on_connection_result` callback
+ *
```

Comment:
may just omit these details about endpoint validation?

I mean, the API contract is still the same: if this function returns 0, then the `on_connection_result` will eventually fire and inform you of whether or not it ultimately succeeded